src/test/java/emissary/util/shell/ExecutrixTest.java

```diff
@@ -80,6 +82,30 @@ public void testExecutrixParams() {
         assertTrue("names must use in file ending", names[Executrix.IN].endsWith(".in"));
     }
 
+    @Test
+    public void testExecutrixUniqueBase() {
+        this.e.setInFileEnding(".in");
+        this.e.setOutFileEnding(".out");
+        this.e.setOrder("NORMAL");
+
+        // Generate 100 sets of names in parallel (multi-threaded) fashion
+        final String[][] names = new String[100][7];
+        IntStream.range(0, 99).parallel().forEach(number -> {
+            names[number] = this.e.makeTempFilenames();
+            assertNotNull("names array has null DIR", names[number][Executrix.DIR]);
+            assertNotNull("names array has null BASE", names[number][Executrix.BASE]);
+            assertNotNull("names array has null BASE_PATH", names[number][Executrix.BASE_PATH]);
+        });
+
+        // Verify that the names generated do not match the next one. This does
+        // not necessarily need to be done in parallel.
+        for (int i = 0; i < 99; i++) {
+            assertNotEquals("DIR matches", names[i + 1][Executrix.DIR], names[i][Executrix.DIR]);
+            assertNotEquals("BASE matches", names[i + 1][Executrix.BASE], names[i][Executrix.BASE]);
+            assertNotEquals("BASE_PATH matches", names[i + 1][Executrix.BASE_PATH], names[i][Executrix.BASE_PATH]);
```

Comment:
Recommend adding ```names[][Executrix.BASE_PATH]``` to a Set and verify that the Set size matches the number of expected unique file paths.

Reply:
Updated testcase to validate unique strings using Sets.